tests/stub/iteration/test_result_peek.py

```diff
@@ -0,0 +1,137 @@
+from contextlib import contextmanager
+
+from nutkit.frontend import Driver
+import nutkit.protocol as types
+from tests.shared import (
+    driver_feature,
+    TestkitTestCase,
+    get_driver_name,
+)
+from tests.stub.shared import StubServer
+
+
+class TestResultPeek(TestkitTestCase):
+    def setUp(self):
+        super().setUp()
+        self._server = StubServer(9001)
+
+    def tearDown(self):
+        self._server.reset()
+        super().tearDown()
+
+    def _assert_not_exactly_one_record_error(self, error):
+        self.assertIsInstance(error, types.DriverError)
+        driver = get_driver_name()
+        if driver in ["python"]:
+            self.assertEqual("<class 'ToBeDecided'>",
+                             error.errorType)
+        else:
+            self.fail("no error mapping is defined for %s driver" % driver)
+
+    def _assert_connection_error(self, error):
+        self.assertIsInstance(error, types.DriverError)
+        driver = get_driver_name()
+        if driver in ["python"]:
+            self.assertEqual("<class 'neo4j.exceptions.ServiceUnavailable'>",
+                             error.errorType)
+        else:
+            self.fail("no error mapping is defined for %s driver" % driver)
+
+    @contextmanager
+    def _session(self, script_fn, fetch_size=2, vars_=None):
+        uri = "bolt://%s" % self._server.address
+        driver = Driver(self._backend, uri,
+                        types.AuthorizationToken(scheme="basic", principal="",
+                                                 credentials=""))
+        self._server.start(path=self.script_path("v4x0", script_fn),
+                           vars=vars_)
+        session = driver.session("w", fetchSize=fetch_size)
+        try:
+            yield session
+            session.close()
+        finally:
+            self._server.reset()
+
+    @driver_feature(types.Feature.API_RESULT_PEEK)
+    def test_result_single_with_0_records(self):
+        with self._session("yield_0_records.script") as session:
+            result = session.run("RETURN 1 AS n")
+            with self.assertRaises(types.DriverError) as exc:
+                result.single()
```

Comment:
I'm not following: shouldn't there be a `peek` call there?
it seems `test_result_peek.py` calls `single` and `test_result_single.py` calls `peek` :)
there is a mix of `@driver_feature` as well